src/components/com_kunena/controllers/user.php

```diff
@@ -452,16 +452,20 @@ public function login()
 			return;
 		}
 
-		$username  = JFactory::getApplication()->input->post->get('username', '');
-		$password  = JFactory::getApplication()->input->post->get('password', '', 'raw');
-		$remember  = JFactory::getApplication()->input->post->get('remember', false);
-		$secretkey  = JFactory::getApplication()->input->post->get('secretkey', null);
+		$app    = JFactory::getApplication();
+		$input  = $app->input;
+		$method = $input->getMethod();
+
+		$username  = $input->$method->get('username', '', 'USERNAME');
```

Comment:
im using the code from joomla itself, I just copied the code. see: https://github.com/joomla/joomla-cms/blob/162d10458cd50387b768f6bf5113836359386b34/components/com_users/controllers/user.php#L39